gradle-plugin/plugin/build.gradle.kts

```diff
@@ -188,3 +186,8 @@ buildConfig {
 tasks.withType<Test> {
   useJUnitPlatform()
 }
+
+tasks.withType<ValidatePlugins>().configureEach {
```

Comment:
This enables stricter validation and requires all our tasks to be marked as either cacheable or not.